src/main/java/com/example/draftspringboot/service/DraftService.java

```diff
@@ -26,14 +28,16 @@ public DraftResponse getById(Integer id) {
         return toModel(draftRepository.getById(id));
     }
 
-    public List<DraftResponse> getAll() {
-        return draftRepository.findAll()
-                .stream()
-                .map(this::toModel)
-                .collect(Collectors.toList());
+    public Page<DraftResponse> searchByPartName(Integer index, Integer limit, String partName) {
+        // index of page can be null
+        // so we will give first (0th) page if index is not set
+        PageRequest pageRequest = PageRequest.of(index == null ? 0 : index, limit);
+        Page<DraftEntity> pageOfEntities = draftRepository.searchByPartName(pageRequest, partName);
+        return new PageImpl<>(pageOfEntities.stream().map(this::toModel).collect(Collectors.toList()),
```

Comment:
You should create the method that accept a Page and return PageModel (PageImpl)

Reply:
Done